### DISCUSSION TITLE
CYBIS connection via Pterm

### DISCUSSION THREAD

**Discussion by robbo007** [kej715/DtCyber#5] [General]
Hi,
I've finally got everything running. In the end running dtcyber on a P100mhz under Sarge was just too slow. I managed to build a P4 machine with Debian 11 (32bit). I can now run dtcyber at a reasonable speed.

I have loaded X.CYBIS. and see the last message on the console on the right hand side showing I have the sub systems loaded. MAS1, PLA1, FOR1, PNI1 and COA1.

Screenshot
[url=https://ibb.co/hXCYKcJ][img][https://i.ibb.co/hXCYKcJ/IMG-9043.jpg[/img][/url]](https://i.ibb.co/hXCYKcJ/IMG-9043.jpg%5B/img%5D%5B/url%5D)

I try connecting from a Mac on my LAN to Cybis via Pterm on port 8005 but I don't get a connection. Is there anything different about the dtcyber program compared to Tom Hunter's original one?

**Comment by kej715** [comment]
Yes, there are differences between Tom's original version of DtCyber and
the NCC fork on GitHub, especially in the area of NPU/MDI configuration.
Have you verified that DtCyber is listening for connections on port 8005
(e.g., "netstat -an|grep LISTEN")? If you use telnet on the DtCyber host to
attempt a connection (e.g., "telnet localhost 8005"), is a connection
established? Sharing a copy of your cyber.ini file  and a copy of your NDL
source file would help to confirm that the CYBIS connections are configured
properly and that the definitions in cyber.ini correctly match the
definitions in the NDL.

**Comment by robbo007** [comment]
ahhh ok that makes sense. I'm using NCC forked dtcyber with Tom's original CybisRelease1 files.  I ran into compiling problems on Debian 11 with Toms original dtcyber so I compiled this one. 

The same CybisRelease1 files I'm using are on: https://codex.sjzoppi.com/doku.php?id=utilities:start

netstat does not show port 8005 and telnet localhost 8005 does not connect. So it looks like its not listen on that port.


Here is my cyber.ini 

`;------------------------------------------------------------------------
;
;   Copyright (c) 2016, Tom Hunter (see license.txt)
;
;
;   Name: cyber.ini
;
;   Description:
;       Define emulation and deadstart parameters.
;
;   Supported OSs:
;       NOS 2.8.7 PSR 871 with CYBIS
;
;------------------------------------------------------------------------

;
; NOS 2.8.7 PSR 871	(CYBIS)
;
[cyber]
model=CYBER865
deadstart=deadstart.cybis
equipment=equipment.cybis
npuConnections=npu.cybis
clock=0
memory=4000000
ecsbanks=0
pps=24
persistDir=PersistStore

[npu.cybis]
; tcp-port,num-conns,type
6610,32,raw
8005,30,pterm
6620,2,rs232

[equipment.cybis]
; type,eqNo,unitNo,channel,path
DD885,0,0,01,Disks/DM01_SYSTFA
DD885,0,0,02,Disks/DM02_SYSTFA
DD885,0,0,03,Disks/DM03_CYBDEV 
DD885,0,0,04,Disks/DM04_BINARY 
DD885,0,1,01,Disks/DM11_DEV0   
DD885,0,1,02,Disks/DM12_CYB0   
DD885,0,1,03,Disks/DM13_CYB1   
DD885,0,1,04,Disks/DM14_CYB2   
DD885,0,2,01,Disks/DM21_UOL    
DD885,0,2,02,Disks/DM22_PUB0   
DD885,0,2,03,Disks/DM23_PUB1   
DD885,0,2,04,Disks/DM24_CYB3   
CO6612,0,0,10
LP512,7,0,07,3555
CR3447,7,0,11
CP3446,6,0,11
MT679,0,0,13,Deadstart/CYBIS_15DS.tap 
MT679,0,1,13
MT679,0,2,13
MT679,0,3,13
TPM,0,0,15
NPU,7,0,05

[deadstart.cybis]
0000
0000
0000
7553 DCN 13
7713 FAN 13, 
0120        0120 for ATS
7413 ACN 13
7113 IAM 13,
7301        7301
0000
0001 wxyy w=level, x=display, yy=cmrdeck
0000

;---------------------------  End Of File  ------------------------------`

**Comment by kej715** [comment]
Your cyber.ini file corresponds to a very old version of DtCyber. It
wouldn't work on Tom's last released version either. In particular, the
[npu.cybis] section was created for a very old version of DtCyber.  It
should probably instead look like:

[cpu.cybis]
terminals=6610,0x01,32,raw
terminals=8005,0x21,30,pterm

**Comment by robbo007** [comment]
Amazing. That resolved it. Thanks :)
